stan/math/fwd/core/std_complex.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef STAN_MATH_FWD_CORE_STD_COMPLEX_HPP
+#define STAN_MATH_FWD_CORE_STD_COMPLEX_HPP
+
+#include <stan/math/prim/core/complex_base.hpp>
+#include <stan/math/fwd/core/fvar.hpp>
+#include <complex>
+#include <type_traits>
+
+namespace std {
+
+/**
+ * Specialization of the standard libary complex number type for
+ * reverse-mode autodiff type `stan::math::fvar<T>`.
+ */
+template <typename T>
+class complex<stan::math::fvar<T>>
+    : public stan::math::complex_base<stan::math::fvar<T>> {
+ public:
+  using base_t = stan::math::complex_base<stan::math::fvar<T>>;
+  using value_type = stan::math::fvar<T>;
+  using complex_type = complex<value_type>;
+
+  /**
+   * Construct complex number from real and imaginary parts.
+   *
+   * @tparam U type of real part
+   * @tparam V type of imaginary part
+   * @param[in] re real part
+   * @param[in] im imaginary part
+   */
+  template <typename U, typename V>
+  complex(const U& re, const V& im) : base_t(re, im) {}
+
+  /**
+   * Constructs complex number from argument, which may be a scalar or
+   * a complex number.  If it is a scalar, the real part is set to the
+   * scalar value and the imaginary part to zero;  if it is complex,
+   * the real and imaginary parts are set to those of the argument.
+   *
+   * @tparam U type of argument
+   * @param[in] x the scalar or complex argument
+   */
+  template <typename U>
+  complex(const U& x) : base_t(x) {}  // NOLINT(runtime/explicit)
```

Comment:
It won't work if it's marked `explicit`.  We need to be able to promote arithmetic types to complex if necessary.

It will only compile if `U` is something assignable to the class template `T`, which means the restriction is to arithmetic types (primitives) or `var` (or to `fvar<U>` for the forward version).  Elsehwere, the decision has been not to tighten down the signature with type traits unless needed for ambiguity resolution.

Reply:
> It won't work if it's marked explicit. We need to be able to promote arithmetic types to complex if necessary.

As is yes but if you added the `operator=` for scalar types it would work right?

I'm fine with it not being explicit but I'll leave this open for now until I have time to sit down and play with this / look at the spec

(also fyi my review is a comment and not a request for changes so everything here is optional)